pkg/chunk/gcp/storage_client.go

```diff
@@ -109,61 +135,61 @@ func (s *storageClient) BatchWrite(ctx context.Context, batch chunk.WriteBatch)
 	return nil
 }
 
-func (s *storageClient) QueryPages(ctx context.Context, query chunk.IndexQuery, callback func(result chunk.ReadBatch, lastPage bool) (shouldContinue bool)) error {
+func (s *storageClientV2) QueryPages(ctx context.Context, query chunk.IndexQuery, callback func(result chunk.ReadBatch, lastPage bool) (shouldContinue bool)) error {
 	sp, ctx := ot.StartSpanFromContext(ctx, "QueryPages", ot.Tag{Key: "tableName", Value: query.TableName}, ot.Tag{Key: "hashValue", Value: query.HashValue})
 	defer sp.Finish()
 
 	table := s.client.Open(query.TableName)
 
-	var rowRange bigtable.RowRange
+	rOpts := []bigtable.ReadOption{
+		bigtable.RowFilter(bigtable.FamilyFilter(columnFamily)),
+	}
+
 	if len(query.RangeValuePrefix) > 0 {
-		rowRange = bigtable.PrefixRange(query.HashValue + separator + string(query.RangeValuePrefix))
+		rOpts = append(rOpts, bigtable.RowFilter(bigtable.ColumnFilter(string(query.RangeValuePrefix)+".*"))) // TODO: Check again and anchor.
```

Comment:
Hmm yeah this is not particularly desirable.  @mbrukman do you know of a better way?

Reply:
@garye, what are your thoughts on this?

Reply:
I'm a bit confused, this is doing a ColumnFilter with a value prefix?

Reply:
Yep. We want to filter by prefix.

Reply:
I was just confused because this is setting up a filter on column names, but the RangeValuePrefix implies that the filter should be used for values instead.

Reply:
This is just a Cortex-ism; our datamodel at this level is super simple, consisting of three-tuples of (hash value, range value, 'cell' value).  We always specify the hash value (mapped to the row key in bigtable), want to do range queries over the range value, and equality filtering on the 'cell' value.

Sorry for the confusion.

Reply:
Makes sense, thanks for the explanation! 

I'm not sure there's a significantly better way to do this... It's probably worth doing this filtering on the bigtable side but, if the amount of data that would get filtered out by this ColumnFilter is small, consider measuring against doing it all client-side.

Reply:
I wonder, would it be valid to apply the same trick @bcotton added to the row ranges; use range query from prefix to prefix+null ?